walkthrough/grapher_cookiecutter/{{cookiecutter.directory_name}}/{{cookiecutter.short_name}}.py

```diff
@@ -3,40 +3,26 @@
 from owid import catalog
 
 from etl import grapher_helpers as gh
-from etl.paths import DATA_DIR
+from etl.helpers import Names
 
+N = Names(__file__)
 
-def get_grapher_dataset() -> catalog.Dataset:
-    dataset = catalog.Dataset(
-        DATA_DIR
-        / "garden"
-        / "{{cookiecutter.namespace}}"
-        / "{{cookiecutter.version}}"
-        / "{{cookiecutter.short_name}}"
-    )
-    assert len(dataset.metadata.sources) == 1
-
-    # short_name should include dataset name and version
-    dataset.metadata.short_name = (
-        "{{cookiecutter.short_name}}__{{cookiecutter.version}}"
-    )
-
-    # move description to source as that is what is shown in grapher
-    # (dataset.description would be displayed under `Internal notes` in the admin UI otherwise)
-    dataset.metadata.sources[0].description = dataset.metadata.description
-    dataset.metadata.description = ""
 
+def get_grapher_dataset() -> catalog.Dataset:
+    dataset = N.garden_dataset
+    # combine sources into a single one and create proper names
+    dataset.metadata = gh.adapt_dataset_metadata_for_grapher(dataset.metadata)
     return dataset
 
 
 def get_grapher_tables(dataset: catalog.Dataset) -> Iterable[catalog.Table]:
     table = dataset["{{cookiecutter.short_name}}"].reset_index()
 
-    # grapher needs a column entity id, that is constructed based on the unique entity names in the database
-    table["entity_id"] = gh.country_to_entity_id(table["country"])
+    # convert `country` into `entity_id` and set indexes for `yield_wide_table`
+    table = gh.adapt_table_for_grapher(table)
 
-    # use entity_id and year as indexes in grapher
-    table = table.set_index(["entity_id", "year"])
+    # optionally set dimensions
+    # table = table.set_index(["dim1", "dim2"], append=True)
```

Comment:
Is it actually necessary to set the index? If not, then it should not even be an option. The only reason I see for this is to verify that the index does not have duplicates (using `verify_integrity=True`, but this is something that should have been done in garden).
On the other hand, if it is necessary to add index, then it could also be done by `adapt_table_for_grapher` (maybe adding arguments `dim1='country'` and `dim2='year'`).

Reply:
Not sure if we're talking about the same thing. I've changed these lines to
```python
# optionally set additional dimensions
# table = table.set_index(["sex", "income_group"], append=True)
```

If you have more these dimensions in multiindex, `yield_wide_table` would create variables grouped by these dimensions (e.g. variables `gdp__sex_male` and `gdp__sex_female`)